libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/location/ConstantVelocityInterpolator.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.ui.maps.internal.location
+
+import android.animation.TimeInterpolator
+import android.graphics.Path
+import android.view.animation.PathInterpolator
+import com.mapbox.geojson.Point
+import kotlin.math.pow
+
+/**
+ * A time interpolator that animates between key points at constant velocity.
+ */
+class ConstantVelocityInterpolator(
+    startPoint: Point,
+    keyPoints: Array<Point>
+) : TimeInterpolator {
+
+    private val innerInterpolator: TimeInterpolator
+
+    init {
+        val distances = mutableListOf<Double>()
+        var total = 0.0
+        keyPoints.fold(startPoint) { prevPoint, point ->
+            val d = distanceSqrTo(prevPoint, point)
+            distances.add(d)
+            total += d
+            point
+        }
+
+        innerInterpolator = if (0 < total) {
+            val path = Path()
+            val step = 1.0f / keyPoints.size
+            var pathTime = 0.0
+            keyPoints.forEachIndexed { index, point ->
+                val deltaTime = distances[index] / total
+                pathTime += deltaTime
+                path.lineTo(pathTime.toFloat(), step * (index + 1))
+            }
+            PathInterpolator(path)
+        } else {
+            TimeInterpolator { it }
+        }
+    }
+
+    override fun getInterpolation(input: Float): Float =
+        innerInterpolator.getInterpolation(input)
+
+    private fun distanceSqrTo(p1: Point, p2: Point): Double {
+        return (p2.latitude() - p1.latitude()).pow(2) +
+            (p2.longitude() - p1.longitude()).pow(2)
```

Comment:
This doesn't look correct, shouldn't it be
```suggestion
        return sqrt((p2.latitude() - p1.latitude()).pow(2) +
            (p2.longitude() - p1.longitude()).pow(2))
```
or basically
```suggestion
        return hypot(p2.latitude() - p1.latitude(), p2.longitude() - p1.longitude())
```

Reply:
Oh @#$%! You are right!!!! My math was wrong. This fixed it. Now works as expected.

https://user-images.githubusercontent.com/2678039/173883490-b525e528-4bbf-44c2-ab16-e302ded44b67.mp4

.

Reply:
Nice, this looks so good right now!

Reply:
Great work putting this together.